src/ring_theory/chain_of_divisors.lean

```diff
@@ -232,11 +259,49 @@ begin
   exact ne_zero_of_dvd_ne_zero hn (subtype.prop (d ⟨c₁ 1 ^ s, _⟩))
 end
 
-variables [decidable_rel ((∣) : associates M → associates M → Prop)]
- [decidable_rel ((∣) : associates N → associates N → Prop)]
+lemma map_prime_of_monotone_equiv [decidable_eq (associates M)]
+  {m p : associates M} {n : associates N} (hn : n ≠ 0) (hp : p ∈ normalized_factors m)
+  (d : {l : associates M // l ≤ m} ≃o {l : associates N // l ≤ n}) :
+  prime (d ⟨p, dvd_of_mem_normalized_factors hp⟩ : associates N) :=
+begin
+  rw ← irreducible_iff_prime,
+  refine (associates.is_atom_iff $ ne_zero_of_dvd_ne_zero hn (d ⟨p, _⟩).prop).mp ⟨_, λ b hb, _⟩,
+  { classical,
+    rw [ne.def, ← associates.is_unit_iff_eq_bot, ← is_unit.factor_order_iso_map_iff _ d],
+    exact prime.not_unit (prime_of_normalized_factor p (by convert hp)) },
+  { obtain ⟨x, hx⟩ := d.surjective ⟨b, le_trans (le_of_lt hb)
+      (d ⟨p, dvd_of_mem_normalized_factors hp⟩).prop⟩,
+    rw [← subtype.coe_mk b _ , subtype.coe_lt_coe, ← hx] at hb,
+    haveI : order_bot {l : associates M // l ≤ m} := subtype.order_bot bot_le,
```

Comment:
shouldn't these be `letI`s as `order_bot` holds data? it may golf the proof a little too